test-framework/core/src/main/java/org/keycloak/test/framework/LogHandler.java

```diff
@@ -0,0 +1,136 @@
+package org.keycloak.test.framework;
+
+import io.quarkus.runtime.logging.LoggingSetupRecorder;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.jboss.logging.Logger;
+import org.jboss.logmanager.LogManager;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.keycloak.test.framework.config.Config;
+
+import java.util.logging.Filter;
+import java.util.logging.Handler;
+
+public class LogHandler {
+
+    private static final Logger LOGGER = Logger.getLogger("testinfo");
+    private final boolean logFilterEnabled;
+
+    public LogHandler() {
+        logFilterEnabled = Config.get("kc.test.log.filter", false, Boolean.class);
+
+        System.setProperty("java.util.logging.manager", "org.jboss.logmanager.LogManager");
+        System.setProperty("junit.quarkus.enable-basic-logging", "false");
+        System.setProperty("log4j2.disable.jmx", "true");
+
+        initializeQuarkusLogging();
+    }
+
+    private static void initializeQuarkusLogging() {
+        ConfigProviderResolver.instance().registerConfig(Config.getConfig(), Thread.currentThread().getContextClassLoader());
+        LoggingSetupRecorder.handleFailedStart();
+    }
+
+    public void beforeAll(ExtensionContext context) {
+        logDivider(Logger.Level.INFO);
+        logTestClassStatus(context, Status.RUNNING, Logger.Level.INFO);
+    }
+
+    public void beforeEachStarting(ExtensionContext context) {
+        logTestMethodStatus(context, Status.INIT, Logger.Level.DEBUG);
+        initLogFilter();
+    }
+
+    public void beforeEachCompleted(ExtensionContext context) {
+        logTestMethodStatus(context, Status.RUNNING, Logger.Level.DEBUG);
+        initLogFilter();
+    }
+
+    public void afterAll(ExtensionContext context) {
+        logTestClassStatus(context, Status.FINISHED, Logger.Level.DEBUG);
+    }
+
+    public void afterEachStarting(ExtensionContext context) {
+        logTestMethodStatus(context, Status.CLEANUP, Logger.Level.DEBUG);
+    }
+
+    public void afterEachCompleted(ExtensionContext context) {
+    }
+
+    public void testSuccessful(ExtensionContext context) {
+        clearLogFilter(false);
+        logTestMethodStatus(context, Status.SUCCESS, Logger.Level.DEBUG);
+    }
+
+    public void testFailed(ExtensionContext context) {
+        clearLogFilter(true);
+        logTestMethodStatus(context, Status.FAILED, Logger.Level.ERROR);
+    }
+
+    public void testAborted(ExtensionContext context) {
+        clearLogFilter(true);
+        logTestMethodStatus(context, Status.ABORTED, Logger.Level.ERROR);
+    }
+
+    public void testDisabled(ExtensionContext context) {
+        clearLogFilter(false);
+        logTestMethodStatus(context, Status.DISABLED, Logger.Level.INFO);
```

Comment:
@stianst Shouldn't it be DEBUG as the success run?

Reply:
Was on the fence on this one, but changing to DEBUG